### ISSUE TITLE
BB to CB migration tool: mcryt to openssl

### ISSUE THREAD

**Issue by Eiskaltereistee** [CircleBilling/CircleBilling#25]

**Comment by namenodeis** [comment]
I have been doing some research on this, actually. Stack Exchange sites are full of different threads where people had this same issue in their own applications on the move to php 7.x

Without mcrypt available, there are some padding hacks that could be done to make openssl do the decryption and then re-encrypt in to openssl standard. The simplest solution would be decrypting with mcrypt and passing it to openssl to re-encrypt. But then users would have to run the script on their old setup first - hardly much of a 'migration' solution.

Here was some useful info, imho. 
https://stackoverflow.com/questions/43329513/upgrading-my-encryption-library-from-mcrypt-to-openssl

I dunno, hadn't got much farther than that, just wishing I could do more to help.

**Comment by Eiskaltereistee** [comment]
It's a very diffecult topic.

I thought about a little Webservice to decrypt -mcryt encrypted- strings in the movement process.
For example:
You start the migration tool, the tool sends all encrypted strings with the salt to our secure webservice.
The webservice decrypt the steings and answers with the encrypted openssl string using the given salt.
Then the migration tool creates a new salt and decrypt the returned data and encrypted that again with the new salt.

But for that we need to discuss about security and agreement stuff..

**Comment by namenodeis** [comment]
I agree it is difficult. But I think setting up a service for it would incur more responsibility than necessary. 

Ideally we should figure out how to decrypt with openssl and create a migration script that would be used in place of the install script. 

I don't mind to put time in to experimenting with the crypto part, but I am a little lost as to where to find what mcrypt *was* doing. If I could figure out how to modify the openssl decryption to match the mcrypt decryption - that is the hardest step, no?

**Comment by Eiskaltereistee** [comment]
You should consider the effort.
A web service needs 2 - 6 hours to get running and doing the stuff we need.

How much time do a developer need to find another solution? Also I don't think it's possible to decrpyt encrypted (With mcrypt) strings with openssl, there are many tries wich are failed. 

Maybe @62BRAINS have an idea about that. 

Important for me is: It need to get working with every system. No extra packages and plugins. I think we don't have that much possibilities.

@namenodeis thank you for your discussion! Very happy to see active members. :)
